server/common/rest.py

```diff
@@ -40,6 +42,41 @@
 from server.common.utils.uns import spatial_metadata_get
 from server.dataset import dataset_metadata
 
+LOCAL_DEV_USER_ID = "local-dev-user"
+RDEV_USER_ID = "rdev-user"
+
+
+HEADER_USER_ID_CANDIDATES = ()
+
+
+def _get_request_user_id(req) -> Optional[str]:
+
+    if req is None:
+        return None
+
+    user = req.headers.get(
+        "X-Auth-Request-User",
+    )
+    if user:
+        return user
+
+
+def _resolve_request_user_id(req) -> Optional[str]:
```

Comment:
Reviewer note: `_resolve_request_user_id` is the gate everything else uses. If `CELLXGENE_DISABLE_AUTH` is set we intentionally bail out (returning 401 on write routes), otherwise we fall back to rdev/dev defaults. The new annotations/genescets endpoints all call this, so keeping the ENV toggles in mind helps make sense of the infra pieces later in the diff.